Add Stripe toolkit

EricGustin opened the pull request:
| Name                              | Description                                                               | Package | Version |
|-----------------------------------|---------------------------------------------------------------------------|---------|---------|
| Stripe.CreateCustomer             | This tool will create a customer in Stripe.                              | Stripe  | 0.0.1   |
| Stripe.ListCustomers              | This tool will fetch a list of Customers from Stripe.                    | Stripe  | 0.0.1   |
| Stripe.CreateProduct              | This tool will create a product in Stripe.                               | Stripe  | 0.0.1   |
| Stripe.ListProducts               | This tool will fetch a list of Products from Stripe.                     | Stripe  | 0.0.1   |
| Stripe.CreatePrice                | This tool will create a price in Stripe. If a product has not already been | Stripe  | 0.0.1   |
| Stripe.ListPrices                 | This tool will fetch a list of Prices from Stripe.                       | Stripe  | 0.0.1   |
| Stripe.CreatePaymentLink          | This tool will create a payment link in Stripe.                          | Stripe  | 0.0.1   |
| Stripe.ListInvoices               | This tool will list invoices in Stripe.                                  | Stripe  | 0.0.1   |
| Stripe.CreateInvoice              | This tool will create an invoice in Stripe.                              | Stripe  | 0.0.1   |
| Stripe.CreateInvoiceItem          | This tool will create an invoice item in Stripe.                         | Stripe  | 0.0.1   |
| Stripe.FinalizeInvoice            | This tool will finalize an invoice in Stripe.                            | Stripe  | 0.0.1   |
| Stripe.RetrieveBalance            | This tool will retrieve the balance from Stripe. It takes no input.       | Stripe  | 0.0.1   |
| Stripe.CreateRefund               | This tool will refund a payment intent in Stripe.                        | Stripe  | 0.0.1   |
| Stripe.ListPaymentIntents         | This tool will list payment intents in Stripe.                           | Stripe  | 0.0.1   |
| Stripe.CreateBillingPortalSession | This tool will create a billing portal session.                          | Stripe  | 0.0.1   |

-------------------------

This PR implements the tools in [stripe-agent-toolkit](https://github.com/stripe/agent-toolkit) verbatim. This means that the logic needed to implement these tools is minimal since stripe has already done the heavy lifting. 

The tools added in this toolkit are the same tools that are in the stripe-agent-toolkit, but formatted as Arcade tools. This means that the names of the tools, the parameter annotations, and tool docstrings are taken from the stripe-agent-toolkit. I have omitted evals since the tool definitions are taken from the stripe-agent-toolkit.

The tools in this PR are generated via a script. I have included this script in the PR (`_generate.py`) so that if the stripe-agent-toolkit ever adds more tools, then we can simply run that script to generate the new tools.

Spartee commented on toolkits/stripe/.ruff.toml:
```diff
@@ -0,0 +1,44 @@
+target-version = "py39"
```
we should start pputting 310

Spartee commented on toolkits/stripe/Makefile:
```diff
@@ -0,0 +1,53 @@
+	@echo "📦 Checking if Poetry is installed"
+	@if ! command -v poetry &> /dev/null; then \
+		echo "📦 Installing Poetry with pip"; \
+		pip install poetry==1.8.5; \
```
do we check for pip?

Spartee commented on toolkits/stripe/pyproject.toml:
```diff
@@ -0,0 +1,41 @@
+ruff = "^0.7.4"
+
+[build-system]
+requires = ["poetry-core>=1.0.0"]
```
and less than 2

Spartee commented on toolkits/stripe/pyproject.toml:
```diff
@@ -0,0 +1,41 @@
+authors = ["Arcade <dev@arcade.dev>"]
+
+[tool.poetry.dependencies]
+python = "^3.11"
```
if 3.11 why lint for 3.9? or 3.10?

Spartee commented on toolkits/stripe/pyproject.toml:
```diff
@@ -0,0 +1,41 @@
+[tool.poetry]
+name = "arcade_stripe"
+version = "0.0.1"
+description = "LLM tools for interacting with Stripe"
```
"Arcade.dev LlM tools"

Spartee commented on toolkits/stripe/tests/test_stripe.py:
```diff
@@ -0,0 +1,50 @@
+    assert dummy_instances[0].secret_key == "test_secret_key"  # noqa: S105
+
+
+def test_run_stripe_tool_with_all_params(monkeypatch):
```
does this parameterize and run through all of them?

EricGustin replied:
it does now!

Spartee reviewed (commented):
Well done. Needs better tests. too many failure scenarios that need to be accounted for with this kind of toolkit. Also .... evals?

EricGustin commented:
@Spartee 

1. I've compiled your general comments on toolkit metadata (target python version for linter, toolkit description, etc.) into a ClickUp ticket & will address them across all toolkits in a separate PR.

2. I updated the tests to at least invoke all tools with all parameter permutations. The logic in this toolkit is quite sparse since it's essentially a wrapper on top of stripe-agent-toolkit, so theirs not much to test.

3. I added some evals.

Spartee commented:
@EricGustin G2G
